tao_compiler/mlir/disc/transforms/input_inline_fusion_pattern.cc

```diff
@@ -81,14 +81,20 @@ bool miscFuseHelper<ConstOp>(PatternRewriter& rewriter, Operation* user,
                              const SmallVector<memref::LoadOp>& load_ops,
                              LowerConfig* lower_config) {
   if (!isa<ConstOp>(producer)) return false;
-  auto memref_type =
-      cast<LmhloOp>(producer).getResultBuffer().getType().cast<MemRefType>();
-  assert(memref_type.getRank() == 0 && "only scalar ConstOp can be fused");
+  auto constant = cast<lmhlo::ConstOp>(producer);
+  auto memref_type = constant.output().getType().cast<MemRefType>();
+  bool is_splat = constant.value().isSplat();
```

Comment:
Does splat const always be scalar const values? is `[2, 3, 4] -> memref<2x3xf32>` valid as well?

Reply:
it is always scalar.
`  let builders = [
    AttrBuilderWithInferredContext<(ins "ShapedType":$type,
                                        "ArrayRef<StringRef>":$values), [{
      return $_get(type.getContext(), type, values,
                   /* isSplat */(values.size() == 1));
    }]>,
  ];`